src/tests/JIT/Intrinsics/TypeIntrinsics.cs

```diff
@@ -258,6 +348,26 @@ static void ThrowsNRE(Action action, [CallerLineNumber] int line = 0, [CallerFil
     }
 }
 
+public class ClassUsingIsGenericTypeOnT<T>
+{
+    public bool IsGenericType() => typeof(T).IsGenericType;
```

Comment:
You may want to mark this as NoInlining to guarantee that it is going to be shared generic code.

Reply:
Fixed in https://github.com/dotnet/runtime/pull/99555/commits/e8e62ea1abbc42b15cc54410f356d0ec00a54c03, thank you!